presto-cli/src/main/java/com/facebook/presto/cli/QueryPreprocessor.java

```diff
@@ -155,7 +167,9 @@ private static String preprocessQueryInternal(Optional<String> catalog, Optional
 
             // check we got a valid exit code
             if (exitCode != 0) {
-                throw new QueryPreprocessorException("Query preprocessor failed:\n\n" + result);
+                Optional<String> errorMessage = tryGetFutureValue(readStderr, 100, MILLISECONDS)
+                        .flatMap(value -> Optional.ofNullable(emptyToNull(value.trim())));
+                throw new QueryPreprocessorException("Query preprocessor failed: " + errorMessage.orElseGet(() -> "Exit code " + exitCode));
```

Comment:
I agree with always printing the exit code.

Maybe we could detect if the error message is multiple lines and treat it separately? For single line, treat it as above, otherwise something like this:
```
Query preprocessor exited with 99:
===
Error message
Blah blah
===